addons/website/static/src/components/views/page_views_mixin.js

```diff
@@ -84,12 +109,20 @@ export const PageRendererMixin = (component) => class extends component {
      *     -> Show all generic/specific records.
      * - A website is selected:
      *     -> Display website-specific records & generic ones (only those without
-     *        specific clones).
+     *        a specific clone on the active website).
+     * This is used on first load only to avoid a flickering effect.
      */
     recordFilter(record, records) {
+        if (!this.props.firstLoad) {
+            return true;
+        }
         const websiteId = record.data.website_id && record.data.website_id[0];
         return !this.props.activeWebsite.id
             || this.props.activeWebsite.id === websiteId
-            || !websiteId && records.filter(rec => rec.data.website_url === record.data.website_url).length === 1;
+            || !websiteId && !records.some(rec => {
+                    const recWebsiteId = rec.data.website_id && rec.data.website_id[0];
+                    return rec.data.website_url === record.data.website_url
+                            && recWebsiteId === this.props.activeWebsite.id;
+                });
```

Comment:
Why do we need to change something here? If I got this right, it's about filtering the list view (and not the counter), which is working already fine, right?
And from what I read in your commit message, this is not related to the counter (which will be wrong briefly), so it's not about that either.

Reply:
Answering this first because it's the easiest and I need to remember the rest:
The filtering is not actually working fine. In case you have a "/contactus" specific to website 1, without the change, if you check the pages of website 2, you won't see the "/contactus" (even though you should have the generic one). Because the previous logic would only consider either pages of website 2, or pages which aren't specific to any website.